kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -96,24 +96,35 @@ def __init__(self, name, init_config, agentConfig, instances=None):
             'kube_service_spec_type': self.count_objects_by_tags,
         }
 
-        # Handling jobs succeeded/failed counts
-        self.failed_job_counts = defaultdict(KubernetesState.JobCount)
-        self.succeeded_job_counts = defaultdict(KubernetesState.JobCount)
+        # Handling cron jobs succeeded/failed counts
+        self.failed_cron_job_counts = defaultdict(KubernetesState.CronJobCount)
+        self.succeeded_cron_job_counts = defaultdict(KubernetesState.CronJobCount)
 
     def check(self, instance):
         endpoint = instance.get('kube_state_url')
 
+        # Logic for Jobs
+        self.job_succeeded_count = defaultdict(int)
```

Comment:
why are these defined in the `check` function and not in the `__init__` ?

Reply:
good point, this is needed if we delete and recreate the job.
Otherwise we will sned 1 again and no change will be done to the counter.

Reply:
I added another test as well to test this use case